homeassistant/components/homewizard/config_flow.py

```diff
@@ -73,6 +89,45 @@ async def async_step_user(
             errors=errors,
         )
 
+    async def async_step_authorize(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Step where we attempt to get a token."""
+        assert self.ip_address
+
+        api = HomeWizardEnergyV2(self.ip_address)
+        token: str | None = None
+
+        # Tell device we want a token, user must now press the button within 30 seconds
+        with contextlib.suppress(DisabledError):
+            token = await api.get_token("home-assistant")
```

Comment:
do we keep requesting this token even if the user clicked submit?

Reply:
Yes; The first request opens the "You can press the button now"-timer. The second request returns a token when the user has pressed the button.

Reply:
More information about this flow can be found here: https://api-documentation.homewizard.com/docs/v2/authorization#create-user-and-get-token